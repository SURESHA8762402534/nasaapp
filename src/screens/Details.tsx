/** @format */

import axios from "axios";
import { ArrowBack, Navigation } from "@material-ui/icons";
import React, { FC, useCallback, useEffect, useState } from "react";
import {
  ActivityIndicator,
  SafeAreaView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { styles } from "../styles/screens";

import { detailsProp } from "../types/screens";

const Details: FC<detailsProp> = (props) => {
  const data = props.route.params.data;

  return (
    <SafeAreaView testID="details" style={styles.container}>
      <View style={styles.card}>
        <Text style={styles.txt}>Name: {data?.name}</Text>
        <Text style={styles.txt}>Nasa JPl URL: {data?.nasa_jpl_url}</Text>
        <Text style={styles.txt}>
          Is potentially hazardous:
          {data?.is_potentially_hazardous_asteroid ? "yes" : "no"}
        </Text>
      </View>
    </SafeAreaView>
  );
};

export default Details;
