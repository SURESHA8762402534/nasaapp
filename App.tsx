/** @format */

import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import Details from "./src/screens/Details";
import Home from "./src/screens/Home";

export default function App() {
  const Stack = createNativeStackNavigator();
  const options = {
    headerShown: false,
  };
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={Home} />
        <Stack.Screen name="`" component={Details} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
