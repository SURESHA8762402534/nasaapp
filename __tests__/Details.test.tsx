/** @format */

import { render } from "@testing-library/react-native";
import React from "react";
import App from "../App";
import Details from "../src/screens/Details";

const props = {
  params: {
    data: "1234",
  },
};

describe("Details", () => {
  jest.useFakeTimers();
  it("renders without crash", () => {
    render(<Details route={props} />);
  });

  it("matches snapshot", () => {
    const tree = render(<Details route={props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
