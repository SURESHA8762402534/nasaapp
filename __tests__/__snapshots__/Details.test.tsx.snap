// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f6f1eb",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
      "width": "100%",
    }
  }
  testID="details"
>
  <View
    style={
      Object {
        "borderColor": "orange",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "padding": 15,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3a6599",
          "fontSize": 16,
        }
      }
    >
      Name: 
    </Text>
    <Text
      style={
        Object {
          "color": "#3a6599",
          "fontSize": 16,
        }
      }
    >
      Nasa JPl URL: 
    </Text>
    <Text
      style={
        Object {
          "color": "#3a6599",
          "fontSize": 16,
        }
      }
    >
      Is potentially hazardous:
      no
    </Text>
  </View>
</RCTSafeAreaView>
`;
